src/software/thunderscope/common/proto_configuration_widget.py

```diff
@@ -58,9 +70,199 @@ def __init__(
         self.param_group.sigTreeStateChanged.connect(self.__handle_parameter_changed)
         self.param_tree.setAlternatingRowColors(False)
 
+        self.save_hbox_top, self.save_hbox_bottom = self.create_widget()
+
+        layout.addLayout(self.save_hbox_top)
+        layout.addLayout(self.save_hbox_bottom)
+
         layout.addWidget(self.search_query)
         layout.addWidget(self.param_tree)
 
+        self.run_onetime_async(3, self.send_proto_to_fullsystem)
+
+    def run_onetime_async(self, time_in_seconds: float, func: Callable):
+        """
+        starting a timer that runs a given function after a given
+        amount of seconds one time asynchronously.
+
+        :time_in_seconds: the amount of time in seconds
+        :func: the function that is going to be ran
+        """
+        self.timer = QTimer()
+        self.timer.setSingleShot(True)
+        self.timer.timeout.connect(func)
+        self.timer.start(time_in_seconds * MILLISECONDS_PER_SECOND)
+
+    def create_widget(self):
+        """
+        Creating widgets that are used to load, save parameters
+
+        :return: the top of the layout, the bottom of the layout, and edit box widget
+        """
+        save_button = QPushButton("Save Parameters")
+        load_proto_button = QPushButton("Load Parameters")
+        reset_button = QPushButton("Reset All Parameters")
+
+        reset_button.clicked.connect(self.reset_button_callback)
+        save_button.clicked.connect(self.save_proto_callback)
+        load_proto_button.clicked.connect(self.load_proto_with_file_explorer)
+
+        save_hbox_bottom = QHBoxLayout()
+        save_hbox_bottom.addWidget(load_proto_button)
+        save_hbox_bottom.addWidget(reset_button)
+        save_hbox_bottom.addWidget(save_button)
+
+        save_hbox_top = QHBoxLayout()
+
+        return save_hbox_top, save_hbox_bottom
+    def update_proto_from_file(self, path_to_file: str):
+        """
+        load the protobuf from path_to_file to the variable self.proto_to_configure
+
+        :param path_to_file:
+        """
+        if not os.path.isfile(path_to_file):
+            self.proto_to_configure = ThunderbotsConfig()
+            self.proto_to_configure.sensor_fusion_config.friendly_color_yellow = (
+                self.is_yellow
+            )
+
+            self.build_proto(self.proto_to_configure)
+            self.save_current_config_to_file(self.path_to_file)
+            return
```

Comment:
We should probably print that no config file was found at path_to_file

Reply:
done.